xcube/core/store/descriptor.py

```diff
@@ -119,6 +129,22 @@ class DatasetDescriptor(DataDescriptor):
     """
     A descriptor for a gridded, N-dimensional dataset represented by xarray.Dataset.
     Comprises a description of the data variables contained in the dataset.
+
+    :param data_id: An identifier for the data
+    :param type_specifier: A type specifier for the data
+    :param crs: A coordinate reference system identifier, as an EPSG, PROJ or WKT string
+    :param bbox: A bounding box of the data (order: min lon, min lat, max lon, max lat)
+    :param spatial_res: The spatial resolution of the data set
+    :param time_range: Start and end time delimiting this data's temporal extent
+    :param time_period: The data's periodicity if it is evenly temporally resolved.
+    :param coords: A list of the dataset's data coordinates
+    :param dims: A mapping of the dataset's dimensions to their sizes
+    :param data_vars: A mapping of the dataset's variable names to VariableDescriptors
+    (``xcube.core.store.VariableDescriptor``). For backwards compatibility, also lists of
+    VariableDescriptors are accepted. This compatibility will come to an end with a future release.
```

Comment:
I'd vote for breaking compatibility now, The framework is new and next release will be a `0.7` anyway. We unfortunatley do not have enough resources to maintain this kind of deprecations.

Reply:
Okay with me. Waiting for @pont-us opinion.